lib/instrument/headers.js

```diff
@@ -0,0 +1,339 @@
+  recv: 1,
+  send: 2,
+  reply: 3
+});
+
+const AUX_TYPES = Object.freeze({
+  NSKeyed: 2,
+  UInt32LE: 3,
+  BigUInt64LE: 4,
+  BigInt64LE: 6
+});
+
+/**
+ * DTXMessageHeader length 32
+ * struct DTXMessageHeader {
+ *     u32 magic DTX_MESSAGE_HEADER_MAGIC
+ *     u32 headerLength
+ *     u16 fragmentId   packet number
+ *     u16 fragmentCount  packets total number
+ *     u32 payloadLength
+ *     u32 identifier
+ *     u32 conversationIndex  +1 indicate a reply message
+ *     u32 channel
+ *     u32 expectsReply  need to return ack data
+ *     }
+ */
+class DTXMessageHeader {
+
+  static build (data) {
+    const messageHeader = Buffer.alloc(DTX_MESSAGE_HEADER_LENGTH);
+    messageHeader.writeUInt32LE(DTX_MESSAGE_HEADER_MAGIC, 0);
+    messageHeader.writeUInt32LE(DTX_MESSAGE_HEADER_LENGTH, 4);
+    messageHeader.writeUInt16LE(0, 8);
+    messageHeader.writeUInt16LE(1, 10);
+    messageHeader.writeUInt32LE(data.payloadLength, 12);
+    messageHeader.writeUInt32LE(data.identifier, 16);
+    messageHeader.writeUInt32LE(data.conversationIndex, 20);
+    messageHeader.writeUInt32LE(data.channel, 24);
+    messageHeader.writeUInt32LE(data.expectsReply, 28);
+    return messageHeader;
+  }
+
+  static parse (headerBuffer) {
+    return {
+      magic: headerBuffer.readUInt32LE(0),
+      headerLength: headerBuffer.readUInt32LE(4),
+      fragmentId: headerBuffer.readUInt16LE(8),
+      fragmentCount: headerBuffer.readUInt16LE(10),
+      payloadLength: headerBuffer.readUInt32LE(12),
+      identifier: headerBuffer.readUInt32LE(16),
+      conversationIndex: headerBuffer.readUInt32LE(20),
+      channel: headerBuffer.readUInt32LE(24),
+      expectsReply: headerBuffer.readUInt32LE(28),
+    };
+  }
+}
+
+/**
+ * DTXMessagePayloadHeader length 16
+ * struct DTXMessageHeader {
+ *     u32 flags
+ *     u32 auxLength  DTXMessageAuxBuffer.length(nullable)
+ *     u64 totalLength  DTXMessageAuxBuffer.length(nullable) + selectorBuffer.length
+ *     }
+ */
+class DTXMessagePayloadHeader {
+  static build (data) {
+    const messageHeader = Buffer.alloc(DTX_MESSAGE_PAYLOAD_HEADER_LENGTH);
+    messageHeader.writeUInt32LE(data.flags, 0);
+    messageHeader.writeUInt32LE(data.auxLength, 4);
+    // eslint-disable-next-line no-undef
+    messageHeader.writeBigUInt64LE(BigInt(data.totalLength), 8);
+    return messageHeader;
+  }
+
+  static parse (headerBuffer) {
+    return {
+      flags: headerBuffer.readUInt32LE(0),
+      auxLength: headerBuffer.readUInt32LE(4),
+      totalLength: Number(headerBuffer.readBigInt64LE(8))
+    };
+  }
+}
+
+
+class DTXMessageAux {
+  constructor (data = []) {
+    this.data = data;
+  }
+
+  build () {
+    const messageAux = new DTXMessageAuxBuffer();
+    if (this.data.length === 0) {
+      return Buffer.alloc(0);
+    }
+    for (const arg of this.data) {
+      if (typeof arg === 'number') {
+        if (Math.abs(arg) < INT32) {
+          messageAux.appendInt(arg);
+        } else if (Math.abs(arg) < INT64) {
```

Comment:
javascript integer numbers can never reach the size greater of [Number.MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER). All numbers larger that that would be cut off unless these are stored as BigInt